xline/src/main.rs

```diff
@@ -233,29 +277,138 @@ async fn read_key_pair(
     Some((encoding_key, decoding_key))
 }
 
+/// Generate a `LevelFilter`
+fn match_level(level: LevelConfig) -> Result<LevelFilter> {
+    match level {
+        LevelConfig::Trace => Ok(LevelFilter::TRACE),
+        LevelConfig::Debug => Ok(LevelFilter::DEBUG),
+        LevelConfig::Info => Ok(LevelFilter::INFO),
+        LevelConfig::Warn => Ok(LevelFilter::WARN),
+        LevelConfig::Error => Ok(LevelFilter::ERROR),
+        _ => Err(anyhow!(format!("Invalid verbosity level: {level:?}"))),
+    }
+}
+
+/// Generates a `RollingFileAppender`
+fn file_appender(
+    rotation: RotationConfig,
+    file_path: &PathBuf,
+    name: &str,
+) -> Result<RollingFileAppender> {
+    match rotation {
+        RotationConfig::Hourly => Ok(tracing_appender::rolling::hourly(
+            file_path,
+            format!("xline_{name}.log"),
+        )),
+        RotationConfig::Daily => Ok(tracing_appender::rolling::daily(
+            file_path,
+            format!("xline_{name}.log"),
+        )),
+        RotationConfig::Never => Ok(tracing_appender::rolling::never(
+            file_path,
+            format!("xline_{name}.log"),
+        )),
+        _ => Err(anyhow!(format!("Invalid rotation config: {rotation:?}"))),
+    }
+}
+
 #[tokio::main]
 async fn main() -> Result<()> {
     global::set_text_map_propagator(TraceContextPropagator::new());
-    let server_args: ServerArgs = ServerArgs::parse();
-    init_subscriber(
-        server_args.jaeger_online,
-        server_args.jaeger_offline,
-        server_args.jaeger_output_dir,
-        &server_args.name,
-        server_args.jaeger_level,
+    let config: XlineServerConfig = if env::args_os().len() == 1 {
+        let config_file = if let Ok(path) = env::var("XLINE_SERVER_CONFIG") {
+            fs::read_to_string(&path).await?
+        } else {
+            include_str!("../../xline_server.conf").to_owned()
```

Comment:
It's not a good default value. For someone use it seriously, some place like `/etc/...` makes more sence.

Reply:
Fixed